utils/proc-macro-server-api/src/lib.rs

```diff
@@ -0,0 +1,25 @@
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+
+pub mod methods;
+
+pub type RequestId = u64;
+
+#[derive(Serialize, Deserialize)]
+pub struct RpcResponse {
+    pub id: RequestId,
+    pub value: serde_json::Value,
+}
```

Comment:
it's impossible to determine whether response is successful or it carries an error

```suggestion
#[derive(Serialize, Deserialize)]
pub struct RpcResponse {
    pub id: RequestId,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub result: Option<serde_json::Value>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub error: Option<ResponseError>,
}
```